packages/ui/src/Integration/IntegrationFlowAddStep.tsx

```diff
@@ -0,0 +1,101 @@
+import * as H from 'history';
+import { Overlay, Popover } from 'patternfly-react';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import './IntegrationFlowAddStep.css';
+
+export interface IIntegrationFlowAddStepProps {
+  showDetails: boolean;
+  forceTooltip?: boolean;
+  addStepHref?: H.LocationDescriptor;
+  i18nAddStep?: string;
+  addConnectionHref?: H.LocationDescriptor;
+  i18nAddConnection?: string;
+}
+
+export interface IIntegrationFlowAddStepState {
+  showTooltip: boolean;
+}
+
+export class IntegrationFlowAddStep extends React.Component<
+  IIntegrationFlowAddStepProps,
+  IIntegrationFlowAddStepState
+> {
+  public static defaultProps = {
+    active: false,
+  };
+
+  public state = {
+    forceTooltip: false,
+    showTooltip: false,
+  };
+
+  public containerRef = React.createRef<HTMLDivElement>();
+
+  constructor(props: IIntegrationFlowAddStepProps) {
+    super(props);
+    this.showTooltip = this.showTooltip.bind(this);
+    this.hideTooltip = this.hideTooltip.bind(this);
+    this.toggleTooltip = this.toggleTooltip.bind(this);
+  }
+
+  public showTooltip() {
+    this.setState({
+      showTooltip: true,
+    });
+  }
+
+  public hideTooltip() {
+    this.setState({
+      showTooltip: false,
+    });
+  }
+
+  public toggleTooltip() {
+    this.setState({
+      showTooltip: !this.state.showTooltip,
+    });
+  }
+
+  public render() {
+    return (
+      <div
+        className={'integration-flow-add-step'}
+        onMouseEnter={this.showTooltip}
+        onMouseLeave={this.hideTooltip}
+        onTouchStart={this.toggleTooltip}
+      >
+        <div
+          className={'integration-flow-add-step__iconWrapper'}
+          ref={this.containerRef}
+        >
+          <div className={'integration-flow-add-step__icon'}>
+            <i className="fa fa-plus-square" />
+          </div>
+        </div>
+        {this.props.showDetails && this.props.children}
+        <Overlay
+          placement="top"
+          rootClose={false}
+          show={this.props.forceTooltip || this.state.showTooltip}
+          target={this.containerRef.current}
+          container={this}
+        >
+          <Popover
```

Comment:
We should consider combining steps and connections in this codebase and not bring forward the separate pages/options like we have in the angular codebase.  It's something we've wanted to do anyways, may as well simplify this and consider doing it here.  Then this whole popover can go away for example :-)

Reply:
So merge steps and connections together? Actually, this is something I had to ask, what are steps and how do they work? Are hard-coded somehow?